framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -558,10 +564,12 @@ private void handleDimensions(
                         dimensions.subList(1, dimensions.size()),
                         (AnnotatedArrayType) type.getComponentType());
             }
-
             AnnotationMirror dimType =
                     getAnnotatedType(dimensions.get(0)).getAnnotationInHierarchy(UNKNOWNVAL);
-            if (!AnnotationUtils.areSameIgnoringValues(dimType, UNKNOWNVAL)) {
+            if (AnnotationUtils.areSameByClass(dimType, IntRange.class)) {
+                dimType = convertIntRangeToIntVal(dimType);
```

Comment:
Wouldn't it be easier to use `ValueCheckerUtils.getValuesFromRange(range, Long.class);` instead of creating a new `IntVal` annotation? That is, move out the `List<Long>` and either initialize it from the IntVal or the IntRange and then have the common logic.